src/main/lombok/org/heigit/ohsome/ohsomeapi/executor/ContributionsExecutor.java

```diff
@@ -66,13 +100,87 @@ public Response count(boolean isUsersRequest) throws Exception {
+      throws UnsupportedOperationException, Exception {
+    return mapRed.aggregateByTimestamp().map(OSMContribution::getContributorUserId).countUniq();
+  }
+  
+  /**
+   * Performs a count calculation for /contributions/count, /contributions/density or
+   * /contribution/latest/count.
+   * 
+   * @param mapRed a MapReducer of OSM contributions
+   * @param isContributionsLatest the boolean value relative to the endpoint /contributions/latest 
+   * @return SortedMap with counts of contributions aggregated by timestamp
+   * @throws Exception thrown by {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapAggregator
+   *         #count() count}
+   * @throws UnsupportedOperationException thrown by
+   *         {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapReducer#aggregateByTimestamp()
+   *         aggregateByTimeStamp}
+   */
+  private SortedMap<OSHDBTimestamp, Integer> contributionsCount(MapReducer<OSMContribution> mapRed,
+      boolean isContributionsLatest) throws UnsupportedOperationException, Exception {
+    if (isContributionsLatest) {
+      MapReducer<List<OSMContribution>> mapRedGroupByEntity = mapRed.groupByEntity();
+      Optional<FilterExpression> filter = processingData.getFilterExpression();
+      if (filter.isPresent()) {
+        mapRedGroupByEntity = mapRedGroupByEntity.filter(filter.get());
+      }
+      return contributionsFilter(mapRedGroupByEntity
+          .map(listContrsPerEntity -> listContrsPerEntity.get(listContrsPerEntity.size() - 1)))
+          .aggregateByTimestamp(OSMContribution::getTimestamp).count();
+    } else {
+      return contributionsFilter(mapRed).aggregateByTimestamp().count();
+    }
+  }
 
+  /**
+   * Filters contributions by contribution type.
+   * 
+   * @param mapRed a MapReducer to be filtered
+   * @return MapReducer filtered by contribution type
+   */
+  private MapReducer<OSMContribution> contributionsFilter(MapReducer<OSMContribution> mapRed) {
+    String contributionType = servletRequest.getParameter("contributionType");
+    if (contributionType == null) {
+      return mapRed;
+    }
+    List<String> contrTypes =
+        Arrays.asList("CREATION", "DELETION", "GEOMETRYCHANGE", "TAGCHANGE", "OTHERCHANGES");
+    contributionType = contributionType.toUpperCase();
+    if (!contrTypes.contains(contributionType)) {
+      throw new BadRequestException(
+          "The contribution type must be 'creation', 'deletion', 'geometryChange', 'tagChange' "
+              + "or 'otherChanges'.");
+    }
+    if (contributionType.equals("OTHERCHANGES")) {
+      return mapRed.filter(contr -> contr.getContributionTypes().isEmpty());
+    } else {
+      var string = new StringBuilder(contributionType);
+      if (contributionType.equals("TAGCHANGE") || contributionType.equals("GEOMETRYCHANGE")) {
+        string.insert(contributionType.length() - 6, '_');
+      }
+      return mapRed.filter(contr -> contr.is(ContributionType.valueOf(string.toString())));
```

Comment:
not sure I like this "creative" implementation too much. I would have probably done it like this:

```
      contributionType = contributionType.toUpperCase();
      if (contributionType.equals("OTHERCHANGES")) {
        return mapRed.filter(contr -> contr.getContributionTypes().isEmpty());
      }
      ContributionType filterType;
      switch (contributionType) {
        case "CREATION":
          filterType = ContributionType.Creation;
          break;
        …
        default:
          throw new BadRequestException("…");
      }
      return mapRed.filter(contr -> contr.is(filterType));
```

IMHO it is more easily readable/understandable. Also it has the benefit that it's not dependent that the input strings correspond to the `ContributionType` enum and avoids potentially thousands of calls to `ContributionType.valueOf()`.

(just a suggestion)